ui/core/src/api/EthbridgeService/EthbridgeService.ts

```diff
@@ -37,23 +36,32 @@ export default function createEthbridgeService({
     return _web3;
   }
 
-  async function approveBridgeBank(account: string, amount: AssetAmount) {
+  async function approveBridgeBankSpend(account: string, amount: AssetAmount) {
+    // This will popup an approval request in metamask
     const web3 = await ensureWeb3();
-    const bridgeTokenContract = await getBridgeTokenContract(
+    const tokenContract = await getTokenContract(
       web3,
-      bridgetokenContractAddress
+      (amount.asset as Token).address
```

Comment:
I found an issue here. What's this `address` supposed to be?

Reply:
I put the bridgeTokenContractAddress in... It asks me for permission to spend `erowan` 🤔 
<img width="472" alt="Screen Shot 2021-01-21 at 2 01 12 PM" src="https://user-images.githubusercontent.com/5235538/105406278-cf69de80-5bf1-11eb-8c1c-d99a33d5226b.png">

Reply:
Yep we need to do this by design. @ElliotFriedman knows more about this

Reply:
So we need to allow the bridgebank contract to spend our tokens on our behalf.

Reply:
erowan is the bridgeToken incase that was not clear.

Reply:
🤔

Reply:
Looks like that tx is authorizing the bridgebank to spend .00001 erowan on your behalf.